### ISSUE TITLE
[BUG] "unexpected close tag" error from saxes when parsing an excel with multiline text

### ISSUE THREAD

**Issue by lmX2015** [exceljs/exceljs#1958]
## 🐛 Bug Report

When trying to read an excel files with "/<br>/" tags (for instance on multi row text within a button), ExcelJS crash with the following error
```
Error: 43:10: unexpected close tag.
at SaxesParser.makeError ( --- node_modules/saxes/saxes.js:410:16)
at SaxesParser.fail (--- node_modules/saxes/saxes.js:422:26)
at SaxesParser.closeTag (--- node_modules/saxes/saxes.js:2019:18)
at SaxesParser.sCloseTag (--- node_modules/saxes/saxes.js:1605:18)
at SaxesParser.write (--- node_modules/saxes/saxes.js:481:36)
at module.exports (---  node_modules/exceljs/lib/utils/parse-sax.js:21:17)

In short this is due to XML code that is accepted by Excel but refused by the saxes parser (typically unclosed tags)

```

Lib version: 3.10.0 or higher (tested up to 4.3.0). Seems to work on 3.9.0, since the issue comes from saxes that was introduced in replacement of sax-js (see comment below).

## Steps To Reproduce
Import this file [bugged.xlsx](https://github.com/exceljs/exceljs/files/7898621/bugged.xlsx)

```javascript
const workbook = new Workbook();
workbook.xlsx.load(file);
```

## The expected behaviour:

File should be loaded without crashing. 

## Related issue

#https://github.com/exceljs/exceljs/issues/1316 previously reported the bug without further details.

## Possible solution
See below

**Comment by lmX2015** [comment]
Just putting here the comment I had on the other issue about the cause of the issue and ways to resolve.

TLDR, issue comes from the saxes library that has been used since 3.10.10 in replacement of sax-js. 

I encountered this issue too with latest release. Problem does not seems to be related to xlsm but rather to malformed xlm such as unclosed <br> tags, which is allowed by excel but not valid xml. 

Digging further, this is due to this PR https://github.com/exceljs/exceljs/pull/1127 (which explains why @vanyadymousky had only this problem starting at 3.10.0.

The PR itself is a migration of the XML parser from [sax-js ](https://github.com/isaacs/sax-js) that has been unmaintained for 6 years to [saxes](url), one of its forks which is faster, supports typescript and is currently maintained.  

The downside is that saxes has a different philosophy than sax regarding what it is supposed to parse. Saxes refuses to parse XML that is not strictly well formatted and throws error in this case, will sax has the "best effort" approach and tries to parse even not strictly valid files. For most XML files it doesn't make any difference, however excel is more permissive in terms of XML formatting, and accept some unclosed tags, including "\<br>\".  If an excel file contains such XML content, parsing with ExcelJS will fail.

I only see three options to fix that:
- reintegrate sax (even if it is used in complement of saxes, for non standard xml strings)
- fork sax to fix vulnerabilities, support typescript, improve speed etc.
- fork saxes to allow non standard XML to be parsed

While I could spend some times on one of these options, happy to have your opinions about it. 

Also for those who run a similar problem and want quick and dirty fix, a simple solution is to update your dependencies to use 3.9.0 an older version of exceljs.

**Comment by KirianCaumes** [comment]
Another solution I have found is to handle (remove) the unclosing tags before opening it with `exceljs`.

A .xlsx file is just a zip containing some .xml files. So if you can identify the unclosing tags, you can remove them like this: Rename .xlsx file to .zip -> Unzip it -> Rename file content needed -> Zip it -> Go back .xlsx.

Example:

```ts
import AdmZip from 'adm-zip'

/**
 * Prepare excel file: Rename to Zip -> Unzip -> Rename file content needed -> Zip folder -> Go back to original format
 * @param filePath filePath
 */
async function prepare(filePath: string): Promise<void> {
    const folderPath = filePath.replace(/\.[^/.]+$/, '')
    const zipPath = `${folderPath}.zip`

    // Rename .xlsx to .zip
    await rename(filePath, zipPath)

    // Unzip content to folder
    const unzip = new AdmZip(zipPath)
    await new Promise((resolve, reject) => {
        unzip.extractAllToAsync(folderPath, true, true, (error: never) => {
            if (error) {
                reject(error)
            } else {
                resolve('')
            }
        })
    })

    // Remove zip file
    await unlink(zipPath)

    /**
     * Get all files recursively
     * @param path path
     */
    const walk = (path: string): Array<string> =>
        readdirSync(path, { withFileTypes: true }).flatMap(file =>
            file.isDirectory() ? walk(join(path, file.name)) : join(path, file.name),
        )

    /** List of all files decompressed */
    const files = walk(folderPath)

    // Apply all transformation needed
    await Promise.all(
        files
            .filter(path => path.endsWith('.vml'))
            .map(async path => {
                // Remove problematic elements
                const content = await readFile(path, 'utf-8')
                const newContent = content.replace(/<br>/g, '')
                await writeFile(path, newContent)
            }),
    )

    // Create a new zip
    const zip = new AdmZip()
    zip.addLocalFolder(folderPath)
    await zip.writeZipPromise(zipPath)

    // Remove folder
    await rm(folderPath, { recursive: true, force: true })

    // Rename .zip to .xlsx
    await rename(zipPath, filePath)
}
```
